src/rgw/rgw_rados.cc

```diff
@@ -7572,7 +7605,8 @@ int RGWRados::copy_obj_data(RGWObjectCtx& obj_ctx,
     }
   }
 
-  ret = processor.complete(etag, mtime, set_mtime, attrs, delete_at);
+  // XXX: need to copy over compression attr and its orig_size here?
+  ret = processor.complete(ofs, etag, mtime, set_mtime, attrs, delete_at);
```

Comment:
i added some code to decode the uncompressed size from the compression attribute here, and included the commit https://github.com/cbodley/ceph/commit/8e0d9a5aa9f73ad91d1aa29f361672ce95231e7f in https://github.com/cbodley/ceph/commits/wip-rgw-compression - feel free to cherry-pick that (it's passing s3 and swift tests)